packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -106,8 +106,9 @@ class VMService {
     this.wsAddress,
     ReloadSources reloadSources,
     Restart restart,
-    CompileExpression compileExpression,
-  ) {
+    CompileExpression compileExpression, {
+    bool sendPeerNotifications = true,
```

Comment:
How does the test break? This seems like a confusing argument to add to the api purely for testing purposes.

Reply:
`peer.sendNotification` throws an error in the test. For the other registered services like hot restart and hot reload, this is not called because it is guarded behind `if (reloadSources != null)` or similar, and in the test we pass null in the VmService constructor for the checked value. We don't need to guard the `flutterVersion` method against anything similar, which is why I added the bool. Open to better ideas if you have some.

Reply:
I would recommend making mockPeer implement the sendNotification instead. As a bonus point you could catch test failures if we are failing to actually register the right calls to sendNotification for all three service listeners registered.

Reply:
done. The only service that will be registered for the test is `flutterVersion` that is the only one that doesn't require other callbacks like `reloadSources`, `restart`, and `compileExpression`